src/hotspot/cpu/riscv/upcallLinker_riscv.cpp

```diff
@@ -260,12 +260,10 @@ address UpcallLinker::make_upcall_stub(jobject receiver, Method* entry,
   arg_shuffle.generate(_masm, as_VMStorage(shuffle_reg), abi._shadow_space_bytes, 0);
   __ block_comment("} argument shuffle");
 
-  __ block_comment("{ receiver ");
-  __ get_vm_result(j_rarg0, xthread);
-  __ block_comment("} receiver ");
-
-  __ mov_metadata(xmethod, entry);
-  __ sd(xmethod, Address(xthread, JavaThread::callee_target_offset())); // just in case callee is deoptimized
+  __ block_comment("{ load target ");
+  __ movptr(j_rarg0, (intptr_t) receiver);
```

Comment:
Hi @JornVernee , Could you please apply following small add-on change for linux-riscv64? As I witnessed build warning with GCC-13. Otherwise, builds fine and the newly-added test/jdk/java/foreign/TestUpcallStress.java is passing.

```
diff --git a/src/hotspot/cpu/riscv/upcallLinker_riscv.cpp b/src/hotspot/cpu/riscv/upcallLinker_riscv.cpp
index 5c45a679316..55160be99d0 100644
--- a/src/hotspot/cpu/riscv/upcallLinker_riscv.cpp
+++ b/src/hotspot/cpu/riscv/upcallLinker_riscv.cpp
@@ -261,7 +261,7 @@ address UpcallLinker::make_upcall_stub(jobject receiver, Symbol* signature,
   __ block_comment("} argument shuffle");

   __ block_comment("{ load target ");
-  __ movptr(j_rarg0, (intptr_t) receiver);
+  __ movptr(j_rarg0, (address) receiver);
   __ far_call(RuntimeAddress(StubRoutines::upcall_stub_load_target())); // loads Method* into xmethod
   __ block_comment("} load target ");

diff --git a/test/jdk/java/foreign/TestUpcallStress.java b/test/jdk/java/foreign/TestUpcallStress.java
index 3b9b1d4b207..40607746856 100644
--- a/test/jdk/java/foreign/TestUpcallStress.java
+++ b/test/jdk/java/foreign/TestUpcallStress.java
@@ -24,7 +24,7 @@
 /*
  * @test
  * @requires jdk.foreign.linker != "FALLBACK"
- * @requires os.arch == "aarch64" & os.name == "Linux"
+ * @requires (os.arch == "aarch64" | os.arch=="riscv64") & os.name == "Linux"
  * @requires os.maxMemory > 4G
  * @modules java.base/jdk.internal.foreign
  * @build NativeTestHelper CallGeneratorHelper TestUpcallBase
```

Reply:
Were you able to reproduce the original issue on RISC-V?

Reply:
Ok, I'll add riscv as one of the target platforms then